### ISSUE TITLE
MenuBar vanishes if View underneath repaints

### ISSUE THREAD

**Issue by tznind** [gui-cs/Terminal.Gui#2238]
**Describe the bug**
Not sure if this is related to #2225 but if you have a `MenuBar` and beneath it a `View` that is updating itself periodically (e.g. a `TableView`) then the repaint of the `TableView` occurs over the top of any open menu.

![menu-no-repaint](https://user-images.githubusercontent.com/31306100/203286061-cb66c7e0-fdf3-46a5-b2fd-3be4b99fdc6a.gif)

**To Reproduce**
1. Add a MenuBar
2. Add a View that fills the space
3. Add a timeout that calls `SetNeedsDisplay` on the big view every 1s
4. Open the menu

**Expected behavior**
Below view not to render over the top of an open menu

 - Version: Nuget Package 1.8.2

**Comment by BDisp** [comment]
I think it's related. Can you test with the #2227 with the same UI please?

**Comment by tznind** [comment]
Yup seems that this PR has also fixed this issue.

![random-letters](https://user-images.githubusercontent.com/31306100/203303296-eed8e944-f78a-48c6-8a7a-7b2bb323bf46.gif)

```csharp
Application.Init ();
var win = new Window ();
var mb = new MenuBar (new []{new MenuBarItem(
new []{
new MenuItem("Click Me","",()=>{ })
}){
Title = "File (F9)"}
});
win.Add (mb);

var txt = new TextView {
Y = 1,
Width = Dim.Fill (),
Height = Dim.Fill (),
AllowsTab = false,
WordWrap = true,
};

win.Add (txt);
Random r = new Random ();
Application.MainLoop.AddTimeout (TimeSpan.FromSeconds (1),
(m) => {
	txt.Text = new string ((char)r.Next (255), 999);
	return true;
});
Application.Run (win);

```

Although in testing this I found that another Exception gets thrown when I click around in TextView while it is being changed... are you able to replicate:? want me to raise an issue?:

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Index was out of range. Must be non-negative and less than the size of the collection. Arg_ParamName_Name
  Source=System.Private.CoreLib
  StackTrace:
   at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException() in /_/src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs:line 101
   at System.Collections.Generic.List`1.get_Item(Int32 index) in /_/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs:line 150
   at Terminal.Gui.WordWrapManager.GetModelColFromWrappedLines(Int32 line, Int32 col) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 855
   at Terminal.Gui.TextView.SetWrapModel(String caller) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 2315
   at Terminal.Gui.TextView.HistoryText_ChangeText(HistoryTextItem obj) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 1398
   at Terminal.Gui.HistoryText.OnChangeText(HistoryTextItem lines) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 714
   at Terminal.Gui.HistoryText.Clear(ustring text) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 722
   at Terminal.Gui.TextView.set_Text(ustring value) in D:\Repos\gui.cs\Terminal.Gui\Views\TextView.cs:line 1498
   at UICatalog.UICatalogApp.<>c__DisplayClass0_0.<Main>b__0(MainLoop m) in D:\Repos\gui.cs\UICatalog\UICatalog.cs:line 75
   at Terminal.Gui.MainLoop.RunTimers() in D:\Repos\gui.cs\Terminal.Gui\Core\MainLoop.cs:line 227
   at Terminal.Gui.MainLoop.MainIteration() in D:\Repos\gui.cs\Terminal.Gui\Core\MainLoop.cs:line 305
   at Terminal.Gui.Application.RunMainLoopIteration(RunState& state, Boolean wait, Boolean& firstIteration) in D:\Repos\gui.cs\Terminal.Gui\Core\Application.cs:line 1165
   at Terminal.Gui.Application.RunLoop(RunState state, Boolean wait) in D:\Repos\gui.cs\Terminal.Gui\Core\Application.cs:line 1145
   at Terminal.Gui.Application.Run(Toplevel view, Func`2 errorHandler) in D:\Repos\gui.cs\Terminal.Gui\Core\Application.cs:line 1355
   at UICatalog.UICatalogApp.Main(String[] args) in D:\Repos\gui.cs\UICatalog\UICatalog.cs:line 78
```

**Comment by tznind** [comment]
Can you please add add `Fixes #2238` to https://github.com/gui-cs/Terminal.Gui/pull/2227 description so that they are both closed by that PR.  I think it makes more sense than just closing this as duplicate but say if you feel otherwise.

**Comment by BDisp** [comment]
> Can you please add add `Fixes #2238` to #2227 description so that they are both closed by that PR. I think it makes more sense than just closing this as duplicate but say if you feel otherwise.

Done.

**Comment by BDisp** [comment]
> Although in testing this I found that another Exception gets thrown when I click around in TextView while it is being changed... are you able to replicate:? want me to raise an issue?:

I changed the test app to have certain sure that this will not fail with menus more bellow:

```cs
			Application.Init ();
			var win = new Window ();
			var mb = new MenuBar (new []{new MenuBarItem (
				new []{
					new MenuItem("Click Me","",()=>{ })
				}){
					Title = "File (F9)"},
				new MenuBarItem (
					new []{
						new MenuItem("Select All","",()=>{ }),
						new MenuBarItem (
							new []{
								new MenuItem("All","",()=>{ }),
								new MenuItem("Selected","",()=>{ })
							}) {
								Title = "Delete"}
					}){
						Title = "Edit"}
				});
			win.Add (mb);

			var txt = new TextView {
				Y = 1,
				Width = Dim.Fill (),
				Height = Dim.Fill (),
				AllowsTab = false,
				WordWrap = true,
			};

			win.Add (txt);
			Random r = new Random ();
			Application.MainLoop.AddTimeout (TimeSpan.FromSeconds (1),
			(m) => {
				txt.Text = new string ((char)r.Next (255), 999);
				return true;
			});
			Application.Run (win);
		}

```

The exception is related with wrap on `TextView` which I already submitted the PR #2215, but seems not fix this issue. Or you open a new issue or link this to the submitted PR to try fix it on the same branch. Thanks.
